### PR TITLE
add vignette with theoretical background

### PR DESCRIPTION
**Pull request by sbfnk**
addresses https://github.com/epiverse-trace/epichains/issues/35 but adding it here first to it can go with the CRAN submission

### CODE REVIEW

File: vignettes/theoretical_background.Rmd [hunk 1]
```diff
@@ -0,0 +1,148 @@
+---
+title: "Theoretical background"
```
Comment (jamesmbaazam) [inline vignettes/theoretical_background.Rmd:2]:
```suggestion
title: "An overview of the theoretical underpinnings for bpmodels"
```

File: vignettes/theoretical_background.Rmd [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+
+For geometric distributed offspring (corresponding to a negative Binomial with $k=1$) this function is given by
+
+$$
+F(L) = \frac{ 1- R_0^{L + 1} } {1 - R_0^{L - 2}}
+$$
+
+In both cases $f(L)$ denotes cumulative mass functions and therefore the probability of observing a chain of length $L$ is therefore $f(L) - f(L - 1)$.
+
+## Gamma-Borel mixture
```
Comment (jamesmbaazam) [inline vignettes/theoretical_background.Rmd:122]:
```suggestion
### Gamma-Borel mixture
```

File: vignettes/theoretical_background.Rmd [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+An alternative to the negative binomial offspring distribution which represents a Poisson-gamma mixture is a Borel-gamma mixture.
+This could represent situations where the variation is not at the _individual level_ but at the _chain level_, i.e. transmission chains is homogeneous but there is heterogeneity between chains.
+In that case, it can be shown that the resulting log-likelihood of chain sizes is
+
+\begin{align}
+\mathrm{LL}(S) = &\log\Gamma(k + S - 1) - \left(\log\Gamma(k) + \log\Gamma(S + 1) \right) \\
+& + (S-1) \log S - k \log \left(S + \frac{R_0}{k}\right)
+\end{align}
+
+## Size and length distributions for arbitrary offspring distributions
```
Comment (jamesmbaazam) [inline vignettes/theoretical_background.Rmd:134]:
```suggestion
## Numerical approximations of chain size and length distributions
```

File: vignettes/theoretical_background.Rmd [hunk 4]
```diff
@@ -0,0 +1,148 @@
...
+
+The analytic likelihoods stated above are included in the package and used when the corresponding offspring distributions are given.
```
Comment (jamesmbaazam) [inline vignettes/theoretical_background.Rmd:136]:
```suggestion

```

File: vignettes/theoretical_background.Rmd [hunk 5]
```diff
@@ -0,0 +1,148 @@
...
+The probability distribution of outbreak sizes from a branching process with a Poisson offspring distribution (Eq. 2.2 in @farrington2003) is a special case of the [Borel-Tanner distribution](https://en.wikipedia.org/wiki/Borel_distribution#Borel%E2%80%93Tanner_distribution) starting with 1 individual.
+An alternative to the negative binomial offspring distribution which represents a Poisson-gamma mixture is a Borel-gamma mixture.
+This could represent situations where the variation is not at the _individual level_ but at the _chain level_, i.e. transmission chains is homogeneous but there is heterogeneity between chains.
+In that case, it can be shown that the resulting log-likelihood of chain sizes is
+
+\begin{align}
+\mathrm{LL}(S) = &\log\Gamma(k + S - 1) - \left(\log\Gamma(k) + \log\Gamma(S + 1) \right) \\
+& + (S-1) \log S - k \log \left(S + \frac{R_0}{k}\right)
+\end{align}
+
```
Comment (jamesmbaazam) [inline vignettes/theoretical_background.Rmd:133]:
```suggestion

The analytic likelihoods stated above are included in the package and used when the corresponding offspring distributions are given.

```

**Review by jamesmbaazam — APPROVED** [review]
Thanks for adding this vignette. I learned a lot while reviewing it and I think it will be insightful to users/developers seeking to understand the theory behind the functionality provided here. 

I made some suggestions, which are mostly a matter of style except the equation for the loglikelihood of the size distribution of a chain with negative binomial offspring, which I think miss-signed one of the terms. I might be wrong, though.

File: vignettes/theoretical_background.Rmd (hunk 4 above)
Comment (sbfnk) [reply vignettes/theoretical_background.Rmd:136]: Have reworded instead of moving.

**Pull request merged**